### PR TITLE
Update ligo version and switch to `require` instead of conditions

### PR DESCRIPTION
**Pull request by konchunas**

### CODE REVIEW

File: contracts/partial/yToken/lendingMethods.ligo [hunk 1]
```diff
@@ -452,20 +432,13 @@ function liquidate(
             s.tokens
           );
 
-          if outstandingBorrowInCU <= liquidateCollateral
-          then failwith("yToken/liquidation-not-achieved");
-          else skip;
-          if borrowerAccount.borrow = 0n
-          then failwith("yToken/debt-is-zero");
-          else skip;
+          require(liquidateCollateral < outstandingBorrowInCU, "yToken/liquidation-not-achieved");
```
Comment (LikoIlya) [inline contracts/partial/yToken/lendingMethods.ligo:436]: Maybe it's duplicate, this check is lost.
```suggestion
require(borrowerAccount.borrow > 0n, "yToken/debt-is-zero");
```

**Review by LikoIlya — APPROVED** [review]

File: contracts/partial/yToken/lendingMethods.ligo (hunk 1 above)
Comment (konchunas) [reply contracts/partial/yToken/lendingMethods.ligo:436]: I removed it intentionally.
There's a check for underflow down the code.

File: integration_tests/build.sh [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #!/bin/sh
 
 mkdir -p integration_tests/compiled
-docker run -v $PWD:$PWD --rm -i ligolang/ligo:0.24.0 compile-contract $PWD/contracts/main/yToken.ligo main > integration_tests/compiled/yToken.tz
+docker run -v $PWD:$PWD --rm -i ligolang/ligo:0.24.0 compile contract $PWD/contracts/main/yToken.ligo  > integration_tests/compiled/yToken.tz
```
[the remark this answers is not in the document]
Comment (konchunas) [reply integration_tests/build.sh:4]: I checked that it works, but good tip nontheless. Thanks!

File: contracts/partial/yToken/lendingMethods.ligo [hunk 3]
```diff
@@ -609,9 +576,7 @@ function priceCallback(
   var s                 : fullStorage)
                         : fullReturn is
   block {
-    if Tezos.sender =/= s.storage.priceFeedProxy
-    then failwith("yToken/sender-is-not-price-feed");
-    else skip;
+    require(Tezos.sender = s.storage.priceFeedProxy, "yToken/sender-is-not-price-feed");
```
Comment (KStasi) [inline contracts/partial/yToken/lendingMethods.ligo:579]: Use `check_permission` instead

File: contracts/partial/yToken/lendingMethods.ligo [hunk 4]
```diff
@@ -634,13 +599,8 @@ function accrueInterest(
...
     else token.isInterestUpdating := False;
 
-    if Tezos.sender =/= token.interestRateModel
-    then failwith("yToken/not-interest-rate-model-address")
-    else skip;
-
-    if borrowRateF >= token.maxBorrowRate
-    then failwith("yToken/borrow-rate-is-absurdly-high");
-    else skip;
+    require(Tezos.sender = token.interestRateModel, "yToken/not-interest-rate-model-address");
```
Comment (KStasi) [inline contracts/partial/yToken/lendingMethods.ligo:602]: Use `check_permission` instead

File: contracts/partial/yToken/lendingMethods.ligo [hunk 5]
```diff
@@ -634,13 +599,8 @@ function accrueInterest(
     then failwith("yToken/interest-update-wrong-state");
```
Comment (KStasi) [inline contracts/partial/yToken/lendingMethods.ligo:599]: In the PR, some of the condition statements are moved to the beginning of the function. This one can also be moved.

File: contracts/partial/yToken/lendingMethods.ligo [hunk 6]
```diff
@@ -419,21 +405,15 @@ function liquidate(
...
-          then failwith("yToken/borrow-id-undefined");
-          else skip;
-
-          if params.collateralToken >= s.lastTokenId
-          then failwith("yToken/collateral-id-undefined");
-          else skip;
-
           if Tezos.sender = params.borrower
           then failwith("yToken/borrower-cannot-be-liquidator")
           else skip;
```
Comment (KStasi) [inline contracts/partial/yToken/lendingMethods.ligo:419]:
```suggestion
          require(Tezos.sender =/= params.borrower, "yToken/borrower-cannot-be-liquidator");
```

File: contracts/partial/yToken/lendingMethods.ligo (hunk 5 above)
Comment (konchunas) [reply contracts/partial/yToken/lendingMethods.ligo:597]: It actually can't since it requires `token` to be retrieved.

**Pull request merged**